src/qibojit/backends/clifford_operations_cpu.py

```diff
@@ -0,0 +1,301 @@
+import numpy as np
+from numba import njit, prange, uint64
+
+
+@njit("b1[:,:](b1[:,:], u8, u8)", parallel=True, cache=True)
+def H(symplectic_matrix, q, nqubits):
+    r = symplectic_matrix[:-1, -1]
+    x = symplectic_matrix[:-1, :nqubits]
+    z = symplectic_matrix[:-1, nqubits:-1]
+    for i in prange(symplectic_matrix.shape[0] - 1):  # pylint: disable=not-an-iterable
```

Comment:
It is also skipped in your aliases above (I mean `r`, `x`, and `z`), so you could just make a view of the whole matrix, and just pass `symplectic_matrix[:-1]` to the function, that instead will operate on the full view.

Reply:
The last row is solely related to measurements, so the measurement gate is the only one that affects it.

Reply:
Ok, then I'd apply the suggestion above (passing `symplectic_matrix[:-1]`) for all the functions (but `_rowsum` and the invoking `M` gate).

Since `state` is the `symplectic_matrix`, you'd just need to trim the last row in `Gate.apply_clifford`:
https://github.com/qiboteam/qibo/blob/cf6246dd00380de80ee4cfcbe7152942ddab25a8/src/qibo/gates/abstract.py#L394-L395
https://github.com/qiboteam/qibo/blob/cf6246dd00380de80ee4cfcbe7152942ddab25a8/src/qibo/backends/clifford.py#L111

Essentially:
```py
    def apply_clifford(self, backend, state, nqubits):
        return backend.apply_gate_clifford(self, state[:-1], nqubits)
```
and override this method in the measurement gate, with the present one
```py
    def apply_clifford(self, backend, state, nqubits):
        return backend.apply_gate_clifford(self, state, nqubits)
```
https://github.com/qiboteam/qibo/blob/cf6246dd00380de80ee4cfcbe7152942ddab25a8/src/qibo/gates/measurements.py#L11